suite-common/wallet-core/src/device/deviceThunks.ts

```diff
@@ -586,3 +586,24 @@ export const failEntropyCheckThunk = createThunk(
         }
     },
 );
+
+type ProcessEntropyCheckResultThunkParams = {
+    device: AcquiredDevice;
+    result: Awaited<ReturnType<typeof TrezorConnect.resetDevice>>;
+};
+
+export const processEntropyCheckResultThunk = createThunk(
+    `${DEVICE_MODULE_PREFIX}/processEntropyCheckResultThunk`,
+    ({ device, result }: ProcessEntropyCheckResultThunkParams, { dispatch }) => {
+        if (result.success) {
+            dispatch(deviceActions.setEntropyCheckResult({ deviceId: device.id, success: true }));
+
+            return;
+        }
+
+        dispatch(notificationsActions.addToast({ type: 'error', error: result.payload.error }));
+        if (result.payload.code === 'Failure_EntropyCheck') {
+            dispatch(failEntropyCheckThunk({ device, error: result.payload }));
+        }
+    },
```

Comment:
nitpick: Wouldn't this be slightly better and consistent with how it was in the previous thunk? No need for empty return in first `if`.
```suggestion
        if (result.success) {
            dispatch(deviceActions.setEntropyCheckResult({ deviceId: device.id, success: true }));
        } else {
            dispatch(notificationsActions.addToast({ type: 'error', error: result.payload.error }));
            if (result.payload.code === 'Failure_EntropyCheck') {
                dispatch(failEntropyCheckThunk({ device, error: result.payload }));
            }
        }
```

Reply:
Good point, if else is better readable in this case. I tend to overuse the early returns :)